Ctrl/Cmd+S Saves Code to localStorage

jakethecoder95 opened the pull request:
So I did try to do as little changes as possible to the current code.  I created a new action creator of type; "saveCode", and a new code-storage-epic; "userManualSaveEpic".  I also refactored "saveCodeEpic" to "saveOnExecuteChallengeEpic".  The event listener is added and removed in the classic/Editor component.

Before this, the users code was getting saved to localStorage when they attempted to submit the challenge. I've just changed it so that they can manually do it as well.  I kept pressing ctrl+s accidently so I thought it would be cool to add this feature. Please let me know if this can be improved in any way.  I am brand new to open source so I apologize if my approach has not been perfect. If you guys like this idea, It would also be cool to add a small alert message to show the user their codes been saved locally. Just thought I'd start with this and see if people like the idea.

beaucarnes committed:
fix: add Api challenges - Spanish

beaucarnes committed:
fix: add Api challenges - Russian

moT01 committed:
feat(guide): add footer link to article source on GitHub

fix/suggested-changes

thecodingaviator commented:
Please remove the lockfile from the PR

jakethecoder95 commented:
Got it. Thanks

On Wed, Mar 13, 2019, 1:02 AM The Coding Aviator <notifications@github.com>
wrote:

> Please remove the lockfile from the PR
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/freeCodeCamp/freeCodeCamp/pull/35583#issuecomment-472317736>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AlYjkGVPLkJh0YqRJqgy78hKo-4hzUTBks5vWLB9gaJpZM4bsjdn>
> .
>

raisedadead commented:
Thanks for the pull request. Can you point me to any issue thread or forum post if this was discussed for more context.

raisedadead commented:
Ignore previous comment. Found it in your explanation in OP:

> Before this, the users code was getting saved to localStorage when they attempted to submit the challenge. I've just changed it so that they can manually do it as well. I kept pressing ctrl+s accidently so I thought it would be cool to add this feature. [...]

jakethecoder95 commented:
> Ignore previous comment. Found it in your explanation in OP:
> 
> > Before this, the users code was getting saved to localStorage when they attempted to submit the challenge. I've just changed it so that they can manually do it as well. I kept pressing ctrl+s accidently so I thought it would be cool to add this feature. [...]

Bouncey commented on client/src/templates/Challenges/redux/code-storage-epic.js:
```diff
@@ -71,17 +78,26 @@ function clearCodeEpic(action$, state$) {
+
+function saveOnExecuteChallengeEpic(action$, state$) {
   return action$.pipe(
     ofType(types.executeChallenge),
```
`ofType` accepts multiple action types so `ofType(types.executeChallenge, types.saveCode)` will run the epic on execute and save, reducing the duplication

Bouncey commented on client/src/templates/Challenges/redux/index.js:
```diff
@@ -142,6 +143,7 @@ export const unlockCode = createAction(types.unlockCode);
 export const disableBuildOnError = createAction(types.disableBuildOnError);
 export const storedCodeFound = createAction(types.storedCodeFound);
 export const noStoredCodeFound = createAction(types.noStoredCodeFound);
+export const saveCode = createAction(types.saveCode);
```
I think `saveEditorContents` is more descriptive. `code` can mean many things given a different context.

Bouncey commented on client/src/templates/Challenges/redux/index.js:
```diff
@@ -239,6 +242,10 @@ export const reducer = handleActions(
       ...state,
       challengeFiles: payload
     }),
+    [types.saveCode]: (state, { payload }) => ({
```
Why update the store? Does the file change?

jakethecoder95 replied:
Yeah this was not necessary.

Bouncey commented on client/src/templates/Challenges/redux/index.js:
```diff
@@ -221,6 +223,7 @@ export const reducer = handleActions(
       ...state,
       challengeIdToNameMap: payload
     }),
+
```
Extra whitespace

raisedadead commented:
@jakethecoder95 thanks a lot for the PR. 

We also need to warn the users that something from the localStorage will be loaded on the editor. This can be a notification.

Also, can you cleanup the PR by squashing your commits. (I know we can do it while merging, but it just makes QA easy, when I pull this down).

jakethecoder95 commented:
@raisedadead Hey, appreciate the feedback.  

What kind of notification do you mean? Like a small alert saying "Saved to localStorage", similar to what codepen does?

raisedadead committed:
chore: make local installations even faster

raisedadead committed:
fix: update client packages

raisedadead committed:
fix: rollback gatsby-remark-prismjs & more updates

raisedadead committed:
fix: warnings about babel corejs config

RandellDawson committed:
fix: use triple backticks around code

ValeraS committed:
chore: update packages

ValeraS committed:
fix(curriculum): throws is not defined

iamrajiv committed:
fixed typos (#35736)

RandellDawson committed:
fix: performed first conversions

RandellDawson committed:
fix: performed first conversions

aditijoshi14 committed:
Fix/update challenges for responsive-web-design-principles spanish (#35915)

aditijoshi14 committed:
Fix/update challenge for responsive web design principles chinese (#35918)

* Fix/update challenges for responsive-web-design-principles chinese

* fix: corrected translation

eamonne committed:
Addition of overview of nginx files and locations. (#30566)

* Overview of nginx files and locations.

* fix: wasn't necessary to add apt-get under CentOS

RandellDawson committed:
fix: added 15 missing solutions (#35745)

wang1998tina committed:
Moved font-awesome challenges to Font Awesome v5.8.1 (#35813)

* update font awesome challenge to font awesome 5

* v5 syntax changed

n1k1c4 committed:
Correctly translated into Russian (#29983)

mfachrulrazy committed:
Add 'sudo' and Step 3 - 4 (#33067)

Add sudo at the first of each syntax line and added Step 3 (firewall adjustment) and Step 4 (check nginx service)

ValeraS committed:
chore(tools): update docker setup to use npm ci

huyenltnguyen committed:
fix(curriculum): Add solutions to Applied Visual Design challenges (#35815)

* fix: add solutions to Applied Visual Design challenges

* fix: remove extra opening style tag

Manish-Giri committed:
fix(curriculum): Add solution to jQuery challenges (#35776)

ojeytonwilliams committed:
fix: make all ways of representing code match blockquotes

ojeytonwilliams committed:
fix: Update challenge docs with new note guidelines (#35929)

thecodingaviator committed:
fix: Moved instruction to proper place and removed assert text in Font Awesome challenges (#35928)

* fix: Moved instruction to proper place and removed assert text

* Add suggestion about making note strong

Co-Authored-By: thecodingaviator <34807532+thecodingaviator@users.noreply.github.com>

QuincyLarson committed:
Clarify curriculum translations (#35921)

* Clarify curriculum translations

* docs: update header for linking to section

Co-Authored-By: raisedadead <1884376+raisedadead@users.noreply.github.com>

* fix: corrected link syntax

mroberts14 committed:
update index.md (#33486)

Provide more information on what nginx is and some basic tutorial links.

LineCCC committed:
fix(guide): binary search tree min/max height (#35938)

We added the correct solution to the guide article related to this code challenge, and deleted the other information which is not related to the code challenge at all.

GarretKemp committed:
Solution and hint added to challenge (#35762)

sil3nthill committed:
bug fix / test of Spread Operator javascript challenge / english  (#35856)

* bug fix test with the Spread Operator

* test of Spread Operator javascript challenge / english / change test and add a helper function

ojeytonwilliams committed:
fix: Add solutions for Sass challenges (#35777)

RandellDawson committed:
fix(curriculum): Improved the tests for Find the length of a String challenge (#35940)

* fix: split tests into two requests

* fix: removed typeof number check

Co-Authored-By: RandellDawson <5313213+RandellDawson@users.noreply.github.com>

* fix: remove unneeded After Test section

RandellDawson committed:
fix: remove comments before tests (#35906)

RandellDawson committed:
fix: rewrote challenge instructions and tests (#35899)

RandellDawson committed:
fix(curriculum): Added test and corrected solution for Restrict Possible Usernames challenge (#35847)

* fix: added test and corrected solution

* fix: removed redundant regex

Co-Authored-By: RandellDawson <5313213+RandellDawson@users.noreply.github.com>

joonjang committed:
New header added (#26525)

* New header added

Went into an abstract level of what determines a brand based on all the self-development books I have been devouring.

* Minor grammar fixes

RandellDawson committed:
fix(curriculum): add new test to use calculated space for Use grid-column to Control Spacing challenge (#35893)

* fix: use calculated space for test

* fix: reworded comment

Co-Authored-By: RandellDawson <5313213+RandellDawson@users.noreply.github.com>

* fix: changed After Test to Before Test

Co-Authored-By: RandellDawson <5313213+RandellDawson@users.noreply.github.com>

DaniellaTodd committed:
Updated solution-Include intro (#34199) (#34486)

Added intro paragraph and further hints.

RandellDawson committed:
fix(curriculum): Increased starting font size to fix a bug on Chrome browser for Sass Apply a Style Until a Condition is Met with @while challenge (#35935)

* fix: increased starting font size for Chrome

* fix: slight verbiage change

Co-Authored-By: RandellDawson <5313213+RandellDawson@users.noreply.github.com>

* fix/remove-extra-space

daniel-j-mac committed:
Fixed block code and syntax highlighting (#29977)

This commit fixes the example code to demonstrate the working of arithmetic operators so that the entire code, from initial comment to ending curly brace for main(), is in a Markdown code block and is syntax highlighted.  Spacing around the include and int initialization was improved as well.

RandellDawson committed:
fix: added all js to end of 3 backticks

ojeytonwilliams committed:
Give language-* code blocks the same style as generic ones

hegdenischay committed:
Fixed syntax error. (#27164)

* Fixed syntax error.

Fixed a syntax error in line 55.
Also corrected the indentation.

* Update index.md

RandellDawson committed:
fix: added better hint and solution (#35746)

moT01 committed:
fix: added semi-colon not in original code

Co-Authored-By: RandellDawson <5313213+RandellDawson@users.noreply.github.com>

Manish-Giri committed:
fix(curriculum): Add solutions to Bootstrap challenges (#35789)

* fix(curriculum): Add solutions to Bootstrap challenges

thecodingaviator committed:
curriculum: Change pre and code to blockquote in Advanced Node and Express challenges (#35713)

* curriculum: Change pre and code to blockquote in Advanced Node and Express challenge

Tiwaripunit committed:
Add changes " Define " to article. (#33357)

aditijoshi14 committed:
fix: update sass challenge - portuguese description (#35436)

aditijoshi14 committed:
fix: update sass challenge - chinese description (#35438)

aditijoshi14 committed:
fix: basic-html-and html5 challenge chinese - test (#35439)

aditijoshi14 committed:
fix: basic-html-and html5 challenge portuguese - test (#35440)

aditijoshi14 committed:
fix: basic-html-and html5 challenge russian - test (#35544)

aditijoshi14 committed:
fix: update sass challenge - russian description (#35545)

ValeraS committed:
fix: remove unnecessary div and props

ojeytonwilliams committed:
fix: remove unnecessary selector, fix inline code and other issues

navjotjsingh committed:
Broken Links in Sum of all odd Fibbonaci Nos Guide (#35911)

Fixed a broken and a missing link in the guide.

RandellDawson committed:
fix: added 22 missing solutions (#35747)

wang1998tina committed:
Fix(guide articles): Developer/case studies links in English (#35771)

* added working links, descriptions, more info links into cse studies

* deleted code of ethics as it is already listed in another page

* changed wording to -expose- as it flows better

lasjorg committed:
feat(guide): add guide for HTML value attribute challenge (#35783)

marcusparsons committed:
Edited previous section and added new sections (#24828)

* Edited previous section and added new sections

Edited previous section to be csharp code base and formatted a bit better.  Also added new sections about more complex topics of for loops

* fix: removed repl.it links

* Update index.md

andrew-ma committed:
better solution to redux spread operator (#35592)

2 step -> 1 step

RandellDawson committed:
[Fix] correct Guide article example for Iterate Through the Keys of an Object with a for...in Statement (#35197)

* fix: added dot btwn [user] and online

* fix: corrected example for other languages

ojeytonwilliams committed:
fix(client): use picture and source to display correct logo (#35939)

moT01 committed:
update restrict possible usernames solution (#35973)

moT01 committed:
fix/astray-period

beaucarnes committed:
fix: Add Api challenges - Chinese translation (#35164)

* fix: Add Api challenges - Chinese translation

* fix: md formatting

* fix: md formatting

beaucarnes committed:
fix: Add Api challenges - Portuguese translation (#35165)

* fix: Add Api challenges - Portuguese translation

* fix: md formatting

ayushjain9501 committed:
Added Swift Implementation of Counting Sort (#35719)

ojeytonwilliams committed:
fix: add test for and slight rewrite to hazards of imperative (#35331)

* fix: add test for and slight rewrite to hazards of imperative

* fix: Resolve merge conflict

* Update curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/understand-the-hazards-of-using-imperative-code.english.md

Co-Authored-By: ojeytonwilliams <ojeytonwilliams@gmail.com>

* fix: Make the only mistake in seed the use of splice

sil3nthill committed:
Define the Head and Body of an HTML Document for challenge solution (#35826)

* Define the Head and Body of an HTML Document for challenge solution

* edit on check the body tag contain h1 and p tags

ojeytonwilliams committed:
fix: Make code blocks look more conventional

joaomfiumari committed:
Update implement-selection-sort.portuguese.md (#21290)

* Update implement-selection-sort.portuguese.md

Improved portuguese translation for selection-sort challenge

* Refactoring markdown titles

Refactoring markdown titles to keep english as default.

JaceFrio committed:
Update to English Translation Consistency (#21319)

Changed all “User Story” to “História do Usuário”

Peres22Kbron committed:
Add modifications in the translated documentation (#23143)

I am a Brazilian Portuguese native speaker. My suggestions are intended to improve readability and overall text comprehension by Portuguese speakers.

MatheusBarem committed:
Minor portuguese translation fixes (#23318)

Portuguese fixes

MatheusBarem committed:
Minor portuguese translation fixes (#23341)

Portuguese translation fixes, terms in portuguese and etc.

MatheusBarem committed:
Minor translate fixes [Portuguese] (#23390)

Titles in Portuguese and fix of terms in English that not be translated to correct understanding of this article.

MatheusBarem committed:
Minor translate fixes [Portuguese] (#23404)

Titles in Portuguese and minor fixes in translation to Portuguese

rpimpao committed:
Fixed the certificate title translation. (#23550)

rpimpao committed:
Update legacy-data-visualization-certificate.portuguese.md (#23561)

diegosiqueir4 committed:
Fixing Portuguese translation & update description (#21824)

Fixing Portuguese translation and update description of "Text Alternative to Images for Visually Impaired Accessibility" in add-a-text-alternative-to-images-for-visually-impaired-accessibility.portuguese.md

liagiu committed:
Improve Portuguese translation for know-when-alt-text-should-be-left-blank (#23583)

sofiabsilva committed:
Translation fixes (#23621)

jefflung committed:
Update add-a-text-alternative-to-images-for-visually-impaired-accessibility.chinese.md (#23914)

As I am a Chinese so I just improve the smoothness of the readability of the content.

ritagomesdasilva committed:
Made changes to the Text (#24311)

Made changes to the text in order to be accurate with Portuguese language and not Brazilian Portuguese

otavioarc committed:
Fix translation problems that difficult reading (#25390)

liagiu committed:
Update Portuguese translation for add-alt-text-to-an-image-for-accessibility (#22739)

igorhideki committed:
Update translation of titles (#26056)

helen414 committed:
Remove extra word and translation (#26310)

remove extra Chinese  word '的' in front of console.log.  Also, there's no need to translate code into Chinese.

sofiabsilva committed:
minor portuguese translation fixes (#23641)

* minor portuguese translation fixes

* fix: removed extra lines

HelioGuilherme66 committed:
Improves Portuguese translation for challenge add-an-accessible-date-picker.portuguese.md (#27433)

HelioGuilherme66 committed:
Improves Portuguese translation for challenge avoid-colorblindness-issues-by-carefully-choosing-colors-that-convey-information.portuguese.md (#27473)

yi-ge committed:
Fix translation error. (#32559)

* Fixed some translation errors.

* Fixed some translation errors.

* Fixed some translation errors.

These errors will cause the code to be unable to run.

* Fixed some translation errors.

These errors will cause the code to be unable to run.

yi-ge committed:
Fix Chinese character error. (#32564)

* Fixed some translation errors.

* Fixed some translation errors.

* Fixed some translation errors.

These errors will cause the code to be unable to run.

* Fix Chinese character error

Peres22Kbron committed:
Update find-characters-with-lazy-matching.portuguese.md (#28001)

A few suggestions to improve readability.

jefflung committed:
Update give-links-meaning-by-using-descriptive-link-text.chinese.md (#24043)

As I am a Chinese, I want to improve the smoothness of the presentation.

kieffwong committed:
change the wordking from '給它` to '采用' (#28239)

'給它` is like google translate,
采用 sounds smoother.

kieffwong committed:
change the wordking from '給它` to '采用' , also the words '不正确格式' &  '描述' (#28245)

improve the translation wording

rafaelmonteiro committed:
Improve check-for-all-or-none.portuguese.md translation (#24123)

Improve translation of `curriculum/challenges/portuguese/02-javascript-algorithms-and-data-structures/regular-expressions/check-for-all-or-none.portuguese.md` and fix wrong Javascript replacements.

HelioGuilherme66 committed:
Improves Portuguese translation for challenge add-a-text-alternative-to-images-for-visually-impaired-accessibility.portuguese.md (#27455)

pjunior94 committed:
Small adjustment in the translation (#32194)

Small adjustment in the translation

CleoAguiar committed:
Add solution required (#35369)

HelioGuilherme66 committed:
Improves Portuguese translation for challenge avoid-colorblindness-issues-by-using-sufficient-contrast.portuguese.md (#28255)

CleoAguiar committed:
Fix subtitle translation (#34433)

* Fix subtitle translation

- add solution

* Apply suggestions from code review

Co-Authored-By: CleoAguiar <cleotaisas@gmail.com>

* fix: added full solution

ziyaointl committed:
Improve Chinese translation in euler-problem-2 (#29927)

*  Improve Chinese translation in euler-problem-2

* Minor translation fix

* Minor translation fix 2

yi-ge committed:
Some keywords should not be translated. (#32544)

Nirajn2311 committed:
fix(challenges): Fix broken links in Chinese challenges (#35126)

thecodingaviator committed:
fix(curriculum): Hide solution from tests in Chinese challenge (#35252)

* fix(curriculum): Remove solution from tests

* fix(curriculum): Hide solution from tests

Nirajn2311 committed:
fix(challenges): Fix broken links in Spanish challenges (#35129)

ageebc137 committed:
Delete quotes around codetags spanish curriculum (#35772)

vowkn committed:
Fix typos, consistent styling w/ English version (#31238)

Also made the whole text have the same level of formality and corrected Google translations.

ageebc137 committed:
Remove quotes around codetags curriculum/challenges/arabic (#35843)

Mo7amm1d committed:
Improving Arabic translation on several "Basic HTML and HTML5" section files (#34908)

* Improving Arabic translation

Improving Arabic translation in say-hello-to-html-elements.arabic.md file

* Translating Description and Improving Arabic Translation.

Translated the Description from the English version. Also Improving the Arabic translation.

* Fixing a mistake in translating the tests.

* Fixing a mistake I made while translating the tests in headline-with-the-h2-element.arabic.md

* Improving the Arabic translation and fixing grammar.

* adding space to (I'm a p tag) in line 10

* fixing bad indentation.

* fixing bad indentation (mistake by me)

* fixing bad indentation (mistake by me)

* Add missing Description and Test text

Adding missing description text and Test text from English version. Also general improvements to the Translation.

* Adding missing description text and Test text 

Adding missing description text and Test text in in uncomment-html.arabic.md from English version. Also general improvements to the Translation.

* Improving the Arabic translation

* Minor updates to the translation

* Overall Enhancement to the translation.

* General translation improvement 

Improving the translation on the Description, Instructions and the Tests.
Separate text to different lines for better readability.

* Improving "Link to External Pages with Anchor Elements" Arabic Translation.

* Matching characters and entities to the English version.

thecodingaviator committed:
fix(curriculum): Hide solution from tests (#35254)

ivangretsky committed:
Fix russian translation for use-the-flex-direction-property-to-make-a… (#33487)

* Fix russian translation for use-the-flex-direction-property-to-make-a-row.russian.md

* Update use-the-flex-direction-property-to-make-a-row.russian.md

SuperSurok committed:
Improved Russian translation. (#34188)

SuperSurok committed:
Improved Russian Translation. (#34189)

Picklebunker committed:
Updated grammar and term consistency (#30694)

ojeytonwilliams committed:
fix(client): Hide monaco-aria-container on Firefox (#35822)

ojeytonwilliams committed:
fix: Sync blockquote, code and pre's color in night mode

cmccormack committed:
docs: moved local docker setup into local setup file (#35975)

raisedadead committed:
fix(donate): add a space in donate text

Picklebunker committed:
Update use-media-queries-to-create-responsive-layouts.chinese.md (#30686)

* Update use-media-queries-to-create-responsive-layouts.chinese.md

* update some translation

Picklebunker committed:
Updated grammar and readability (#30658)

* Updated grammar and readability 

Terms such as class and item are either referred to in English or avoided in the Chinese versions of tutorials that I have found, as translating them directly does not make much sense.
Changed some sentence structures for the ease of understandings.

* fix typo

Picklebunker committed:
Updated grammar and consistency of terms. (#30668)

Commit:
Update index.md (#30363)

Commit:
Update index.md (#30378)

Commit:
Update index.md (#30386)

Commit:
Update index.md (#30390)

Commit:
Update index.md (#30410)

raisedadead committed:
docs: remove snyk badge

moT01 committed:
Fix/manage state locally tests (#35980)

* fix/manage-state-locally-test

* fix/update-test-words

* fix/update-test

Commit:
Change "rest operator" to say "rest parameter" in challenges and guide (#35496)

* change es6 challenges rest operator to param

* fix rest parameter typos

* change rest operator to parameter in guide

* fix casing

* change rest operator to rest parameter in guide

* change rest operator to rest parameter in curriculum

* remove extra whitespace

* remove whitespaces

* remove whitespace

* fix: removed arabic file

* fix: removed chinese file

MargareV committed:
Update index.md (#30256)

waseemahmad31 committed:
fix NaN issue for the number of active Donations (#35970)

* fixed NaN issue for the number of active Donations

* Initialized activeDonations to 0 in the initial state

ojeytonwilliams committed:
fix(client): Donate other display correct submit button text

kylesureline committed:
Improve formatting of Challenge Seed (#35854)

duytrinhvn committed:
Fix spelling error (#35925)

* Fix spelling error

* fix: changed be conformed to conform

alfredojry committed:
Fix "ReferenceError: url is not defined" in challenge (#35415)

* Fix "ReferenceError: url is not defined" in challenge

I fix the challenge because issue #14002 
- The `url` variable is set to external resource `https://jsonplaceholder.typicode.com/posts`
- I replaced .textContent instead of .innerHTML (better practice)
- Checking for HTTP status of 201 instead of 200
All these suggestions by @HarplingeTom and "approved" by  @QuincyLarson. See the issue comments.
I test the code locally in my computer and this works :+1: 
I modified and verify the tests according the new assessment.

* Update curriculum/challenges/english/04-data-visualization/json-apis-and-ajax/post-data-with-the-javascript-xmlhttprequest-method.english.md

Minor typo: `<code>` to `</code>`

Co-Authored-By: yoelvisj <43357527+yoelvisj@users.noreply.github.com>

* Update curriculum/challenges/english/04-data-visualization/json-apis-and-ajax/post-data-with-the-javascript-xmlhttprequest-method.english.md

new declared as const

Co-Authored-By: yoelvisj <43357527+yoelvisj@users.noreply.github.com>

RandellDawson committed:
fix: removed guide articles no longer needed (#35679)

raisedadead committed:
docs: fix emoji on the translate item

RandellDawson committed:
fix: improved existing tests (#35801)

RandellDawson committed:
Hints & Solution for React and Redux (#35946)

amolpednekar committed:
Generalized smart contract definition | Updated Blockchain technology section (#30994)

Earlier description seemed to Ethereum specific, hence updated.
Also would be more accurate to say Hyperledger - (Specific project) instead of the umbrella term just "Hyperledger" because they have tons of projects.

lipis committed:
Replace Github -> GitHub (Spanish) (#35182)

prashantchikhalkar committed:
add "Quit sftp." option (#29454)

* add "Quit sftp." option

Added "Quit sftp." and manpage referernce

* fix: changed unix to bash

prashantkumar1997 committed:
Added reference links section (#29593)

* Added reference links section

* fix: changed c++ to cpp

* fix: changed one extra c++ to cpp

JustinCThomas committed:
Add a section on Decentralized Apps (#31443)

race2infinity committed:
Added escape sequences, operators in "terms-to-know-for-beginners" (#28832)

* Added escape sequences, operators in "terms-to-know-for-beginners"

Added extra details/facts about multi-line comments in C++
Also fixed some indentation

* fix: changed c++ to cpp

sophiekamuf committed:
Add "K-means clustering description" (#24684)

codyteo committed:
Elaboration of SAFe configurations (#24838)

Elaborating on Full SAFe, Large Solution SAFe, Portfolio SAFe, and Essential SAFe.

Tess2016 committed:
Added Go to list (#31469)

jagzmz committed:
Added "Intent Services" (#30876)

Added more information in services about intent services.

Trion129 committed:
Graph Algorithms: Added ford fulkerson algorithm for maximum flow (#24481)

* added ford fulkerson algorithm guide

* fix: changed to 2 space indentation

Catherinesjkim committed:
Fixed some typos and grammatical errors (#19686)

samujjal1618 committed:
Improved on the Dynamic array description (#27761)

* Improved on the Dynamic array description

* fix: changed c++ to cpp

chris-steenkamp committed:
Add examples of creating static and dynamic arrays (#30358)

* Add examples of creating static and dynamic arrays

* fix: changed c++ to cpp

Cesar-Jim committed:
Adding 1 accent to a word (#22618)

1 instance of the word exito was not accentuated in the article. It should read éxito.

patel-himanshu committed:
Minor grammatical corrections. (#31228)

AndrewPRohde committed:
Changed preform to perform (#31731)

rjanecek committed:
Fixed typo, improved readability. (#32580)

ritaban-datta committed:
Updated recursion (#32505)

Deleted an extra white space

Shooflower committed:
Corrected grammar/spelling errors (#31402)

nitin7ind committed:
Simplified language. (#32814)

Simplified the language, removing duplicate phrases.
Also did little formatting.

PJURC committed:
Prime number calculator page expansion (#32132)

* Prime number calculator page expansion

* fix spacing

jello7654 committed:
clarify text (#34037)

added the to sentences and clarified a question.

jack-chapman committed:
Add information about key binding (#28209)

* Add information about key binding

Also improved grammar and accuracy of information

* fix: removed =====

* Update index.md

* Update index.md

EziamakaNV committed:
Making the code more readable (#29465)

marektro committed:
Expand explanation. (#33890)

sah-pinocchio committed:
add the text in definition of Array. (#30662)

* add the text in definition of Array.

add the text " by adding an index to a unique identifier" after "An array is a series of elements of the same data type which are stored in contiguous memory locations and can be referenced individually".

* fix: changed c++ to cpp

UberschallSamsara committed:
Added section "Relationship to pointers", fixed misspelled "Initializiation" (#31683)

* Added section "Relationship to pointers"

* fix: changed c++ to cpp

chirayusamarth committed:
added info smart_open module (#25000)

frankschmitt committed:
added some basic information and examples for Elm lists (#25042)

condor-webdev committed:
Create tar file (#25056)

dexrualo committed:
Add: GnuPG Guide (#25060)

* Add: GnuPG Guide

* Update index.md

* Update index.md

* Update index.md

* fixed title

GreatDeveloper66 committed:
Expand Propositional Logic Stub (#25398)

This is a simple addition to this stub that briefly explains Modus Ponens.

clylo committed:
Add plots to R guide (#25444)

* Add plots to R guide

* Deleted plotting file

* added a plotting index.md for R

RandellDawson committed:
fix: changed to h4 > s in selector (#36023)

aliciaphes committed:
Fix some typos and refine some expressions in Spanish (#27485)

* Fix some typos and refine some expressions in Spanish

* Making changes as requested from non-Spaniard

* Some more corrections for Spanish

droxine committed:
Minor fixes in the Translations (#32979)

Sobretodo la falta de tildes.

Kaylen7 committed:
Corrected CONTRIBUTING.md Spanish version (#31010)

- Standarised the whole text in the use of 2nd person singular (tú) for referring to the reader. 
- Fixed some discrepancies with capital letters. Usually in Spanish it's not so common to capitalise common nouns. 
- Changed a couple of sentences that didn't sound clear. I used the English version of the article to find the meaning and propose a change of words. 

I honestly don't know how we should translate the word issues. I understand "problema" is definitely it, but in sentences it might not seem clear when it's about a doubt the user has, or when, something reported. I think the usage of the verb "reportar" might be key to clarify the meaning in sentences where both kinds of issues are mixed. (I apologise beforehand if this explanation seems a bit poor. I've been trying to find the right language forum for this kinds of issues ;) ).

Cesar-Jim committed:
Fixing title and one sentence (#22601)

Title read: Código de Documentación
Should read: Documentación del código

theogoulart committed:
Fix translation in Spanish README.md (#21848)

Translate title

Cesar-Jim committed:
Typo fixed (#22562)

* Typo fixed

Changed the title from "Liza" to "Listas"

* Update index.md

aingerusanchez committed:
Code comments translated (#21909)

* Code comments translated

Traducidos los comentarios del código de los ejemplos.

* Update index.md

saeg2k3 committed:
enhancing the translation to spanish (#34064)

fixed: translations
fixed: spell check
fixed: grammar check
removed: removed additional paragraphs non-existent in the original document

icarlossz committed:
Correcting translate errors in contributing guide (Spanish) (#35005)

GaxysGalactic committed:
Add "Find out more" links and section on BLAST (#24949)

Useful links for finding out more on Computational Genomics, and added a section detailing BLAST, a very useful search algorithm and database for bioinformatics.

anthonyjvoss committed:
Added stderr & stdout redirection examples (#28153)

hegdenischay committed:
Fixed Formatting. (#27144)

Fixed the formatting and grammar. 
Also made it clear that `malloc()` is generally discouraged for use in C++. See [here](https://stackoverflow.com/questions/45105100/when-and-why-we-must-use-the-malloc) for more info.

Logan-47 committed:
Added extra commands that comes very handy (#27195)

* update index.md

Added extra commands that comes very handy

* Update index.md

* Update index.md

* fixed a typo

jaqw committed:
This method also cleanup the resources such as threads so that application runs smooth. (#32340)

redsonjacode committed:
Update index.md (#27230)

Edited for grammatical errors and some awkward sentences/phrases. Hope this is helpful.

Descritorio committed:
Added extra information to "Article Tag" section (#34101)

* Added extra information to "Article Tag" section

* Fixed "Article Tag" section

Fixed grammatical errors and combined multiple lines

Vinicius-Targa committed:
Translated lines 22 - 55 (#33745)

Updated the language to portuguese in lines 22 to 55

HKuz committed:
fix(guide): remove redundant Jupyter NB material and passive voice, simplify exp… (#32940)

* fix(guide): remove redundant material and passive voice, simplify explanations

* Text edit to restart build

Mo1914 committed:
Scrum Components (#27581)

Edited Scrum component description and added components.

Commit:
Fix incorrect proper name translation and phrases with wrong meaning (#22950)

Some occurrences of proper names, like "React" and "React Native" were wrongly translated to Portuguese.

KitanGarcia committed:
fix(guide): typos in bash-redirection (#29279)

yash committed:
Fixed link, document format, and grammar (#28023)

adityaarakeri committed:
Adding new commands to the list (#28206)

Adding new commands to the list of commands

erichodges committed:
Added (GUI) to make text more clear (#31739)

Added (GUI) to first line in the first paragraph.

mikejgurch committed:
Added another example to defining-functions article (#28449)

Added an example to help show how to define and call functions in a simple and easy to understand manner.

NNChawla committed:
Python 3 Change (#28596)

Included difference in python 3 vs python 2

kellyhuang21 committed:
Added b-tree insertion (#30195)

* Added b-tree insertion

* Added Insertion

25Harry committed:
Created required brackets in logn (#33401)

omerharshit committed:
Adding Time Analysis for B-Trees (#31013)

* Adding Time Analysis for B-Trees

Time Analysis for B-Tree:

Suppose a B-tree has n elements and M is the maximum number of children a node can have. What is the maximum depth the tree could have? What is the minimum depth the tree could have?

    The worst-case depth (maximum depth) of a B-tree is: logM/2 n.
    The best-case depth (minimum depth) of a B-tree is: logM n.

Worst-Case Times for B-Trees:

    Adding or removing an element in a B-tree with n elements is O(log n).

* fix: removed duplicate info

1nathanliang committed:
corrected typos and clarified sentence structures (#32947)

Joesepherus committed:
Insertion (#30223)

dj-m committed:
Add formatted string literals example (#33192)

Glafs committed:
Added text to boolean-operations in python guides (#31023)

Added descriptions and examples of other boolean-operations which are not part of the Python language, but still useful to know.

iTanuj committed:
added more common and efficient data structures (#29225)

mayank6 committed:
Added a way to get multiple inputs (#31684)

TanayKulkarni committed:
Added python package installer details (#31594)

* Added python package installer details

* Update index.md

mikejgurch committed:
Added examples for complex numbers to numeric-types article (#28443)

* Added examples for complex numbers to numeric-types article

Introduce 3 examples of what a complex number would look like in Python.

* fix: moved links to bottom

AgentMcBride committed:
update index.md (#32087)

Added a way to practice terminal skills with the use of overthewire war-games.

moT01 committed:
fix/record-collection (#36002)

* fix/record-collection

* fix/remove-old-assert-messages

* fix/attempt-to-fix-build-error

* fix/attempt-to-fix-build-error

* fix/change-verbiage-and-tidy-up

* fix/remove-brackets-in-regex

elcodex committed:
[Ru] Improved the translation of Project Euler 02 (#34153)

changed the file "problem-2-even-fibonacci-numbers.russian.md"

clandau committed:
fix to spelling error and add period (#36034)

clandau committed:
fix to spelling error "instantiated" (#36032)

* fix to spelling error

* Update curriculum/challenges/english/06-information-security-and-quality-assurance/advanced-node-and-express/set-up-the-environment.english.md

Co-Authored-By: Randell Dawson <5313213+RandellDawson@users.noreply.github.com>

benfieldta committed:
Grammar fixes (#33049)

Evaldas22 committed:
Add information about pattern matching (#24400)

Added information about how we can use a switch statement for pattern matching and gave an example from Microsoft docs.

huyenltnguyen committed:
Remove tag characters (#36037)

elcodex committed:
[Rus] Improved the translation of Redux challenges (part 3) (#34353)

* Improved the translation

* Improved the translation

* Improved the translation

* Improved the translation

* Improved the translation

elcodex committed:
[Rus] Improved the translation of Redux challenges (part 2) (#34324)

* [Rus] Improved the translation

Changed file "get-state-from-the-redux-store.russian.md"

* Changed line 11

* Improved the translation

* Improved the translation

* Added the translation to sections 

also improved the translation of the sections "tests" and "challengeSeed"

* Improved the translation

Commit:
Added missing `</code>` in test text string (#36040)

RandellDawson committed:
fix(curriculum): Convert blockquote elements to triple backtick syntax for Responsive Web Design (#35993)

* fix: converted blockquotes to code fences

RandellDawson committed:
fix(curriculum): Convert blockquote elements to triple backtick syntax for Information Security And Quality Assurance (#35997)

* fix: converted blockquotes

* fix: added extra line before triple backtick syntax

Co-Authored-By: Oliver Eyton-Williams <ojeytonwilliams@gmail.com>

* fix: corrected misc issues

* fix: properly closed em element

Co-Authored-By: Oliver Eyton-Williams <ojeytonwilliams@gmail.com>

RandellDawson committed:
fix(curriculum): Convert blockquote elements to triple backtick syntax for Apis And Microservices (#35996)

* fix: converted blockquotes

* fix: revert to blockquote

* fix: changed js to http

Co-Authored-By: Oliver Eyton-Williams <ojeytonwilliams@gmail.com>

* fix: reverted back to blockquote

* fix: reverted back to blockquote

* fix: reverted back to blockquote

RandellDawson committed:
fix(curriculum): Convert blockquote elements to triple backtick syntax for Front End Libraries (#35994)

* fix: converted blockquotes for react/redux

* fix: converted blockquotes for sass

* fix: changed jsx to js

Co-Authored-By: Oliver Eyton-Williams <ojeytonwilliams@gmail.com>

* fix: changed jsx to js

Co-Authored-By: Oliver Eyton-Williams <ojeytonwilliams@gmail.com>

RandellDawson committed:
fix(curriculum): Convert blockquote elements to triple backtick syntax for Data Visualization  (#35995)

* fix: convert data visualization

* fix: reverted to blockquote

* fix: changed js to json

Co-Authored-By: Oliver Eyton-Williams <ojeytonwilliams@gmail.com>

* fix: cleaned up code

Co-Authored-By: Oliver Eyton-Williams <ojeytonwilliams@gmail.com>

ojeytonwilliams committed:
feat(client): Add support question covering browser extensions

k4iru committed:
fixed typos in custom css-variable challenge #36038 (#36041)

* fixed typos in custom css-variable challenge

* fix: Remove extra line

weihungchin committed:
Add new example (#30133)

Added example of using map operator to get a list of object properties from a list of objects

Commit:
Nearer specified collect function (#32430)

Added an example to specify how collecting works

J4unty committed:
Java stream additions (#33003)

* Reworked Java Streams

* Added Stream  examples for parallel streams

don0t committed:
Full text editing (#21326)

Changed tone to less formal throughout the text. Grammar editing. Changed a few words and phrases. Improved translation.

code-v1 committed:
Translated HTML comment (#23848)

Translated this text "You can comment out a large number of lines like this." to this "Puedes commentar varias lineas como estas."

alaminos committed:
Some changes in first parragraphs (#24527)

maxmiau committed:
Update python name in index.md (#28360)

* Update index.md

Programming languages has no formal translation to spanish, they are named with their original names.

* fix: changed node to js

code-v1 committed:
Translated html comment (#23860)

Translated this " FreeCodeCamp is an open-source project that needs your help" to this "FreeCodeCamp es un projecto de codigo abierto que neccesita tu ayuda"

RandellDawson committed:
fix: declared x variable (#36046)

RandellDawson committed:
fix: reordered content and added headers (#34713)

thecodingaviator committed:
Update index.md (#34282)

RandellDawson committed:
fix(curriculum): Rewrote the Write Higher-Order Arrow Functions challenge and moved to Functional Programming section (#35954)

* fix: reworded and moved challenge

* fix: removed extra line

Co-Authored-By: RandellDawson <5313213+RandellDawson@users.noreply.github.com>

* fix; removed extra line

Co-Authored-By: RandellDawson <5313213+RandellDawson@users.noreply.github.com>

* fix: reworded sentence

Co-Authored-By: RandellDawson <5313213+RandellDawson@users.noreply.github.com>

RandellDawson committed:
fix(guide) Replace invalid prism code block names (#35961)

* fix: replace sh with shell

fix replace terminal with shell

fix replace node with js

fix replace output with shell

fix replace cs with csharp

fix replace c++ with cpp

fix replace c# with csharp

fix replace javasctipt with js

fix replace syntax  with js

fix replace unix with shell

fix replace linux with shell

fix replace java 8 with java

fix replace swift4 with swift

fix replace react.js with jsx

fix replace javascriot with js

fix replace javacsript with js

fix replace c++ -  with cpp

fix: corrected various typos

fix: replace Algorithm with nothing

fix: replace xaml with xml

fix: replace solidity with nothing

fix: replace c++ with cpp

fix: replace txt with shell

fix: replace code with json and css

fix: replace console with shell

rswalve committed:
Update index.md (#32849)

jakethecoder95 committed:
Notifying user on content save

raisedadead commented:
> What kind of notification do you mean? Like a small alert saying "Saved to localStorage", similar to what codepen does?

Yes, a toast would be nice.

raisedadead commented:
Hi @jakethecoder95 I am going to have to close this PR. Looks like the rebase got really weird.

We would love you to still open a fresh PR if you have the commits on a branch safe somewhere?
